clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/api/v1/ConsulCatalog.groovy

```diff
@@ -26,11 +28,14 @@ import java.util.concurrent.TimeUnit
 class ConsulCatalog {
   public CatalogApi api
 
-  ConsulCatalog(String serverBaseUrl, Long timeout) {
+  ConsulCatalog(ConsulConfig config) {
+    if (!config.enabled) {
```

Comment:
The bodies of the constructor in this class and the next are identical except for the arg to `.create()`. Can the common code be extracted somewhere? Maybe they should share a common base class and then put the common code in the parent class's constructor?